ports/renesas-ra/machine_uart.c

```diff
@@ -557,6 +567,20 @@ STATIC mp_uint_t machine_uart_ioctl(mp_obj_t self_in, mp_uint_t request, uintptr
         if ((flags & MP_STREAM_POLL_WR) && uart_tx_avail(self)) {
             ret |= MP_STREAM_POLL_WR;
         }
+    } else if (request == MP_STREAM_FLUSH) {
+        // The timeout is estimated using the buffer size and the baudrate.
+        // Take the worst case assumptions at 13 bit symbol size times 2.
+        // t.b.d. get actial TX buffer size
+        uint32_t timeout = mp_hal_ticks_ms() +
+            (uint32_t)(uart_tx_txbuf(self)) * 13000ll * 2 / self->baudrate;
+        do {
+            if (machine_uart_txdone((mp_obj_t)self) == mp_const_true) {
```

Comment:
Why not just call `!uart_tx_busy(self)` directly?  That will be a tiny bit faster, and not require a cast.

If you need to cast, you should use `MP_OBJ_FROM_PTR(self)` (that's compatible with nan-boxing mode).

Reply:
Thanks for the hint. Done. I'll re-test and update.

Reply:
The changes are tested and ammended to the commit, which is updated.

Reply:
Excellent!